knowledge_base/Environment-Variables.md

```diff
@@ -0,0 +1,401 @@
+
+## MAGIC_SUPPORTED_BASES
+
+Chain bases supported for Magic login; as of 231212, we use  `substrate,ethereum`.
+
+## MIXPANEL_DEV_TOKEN
+
+Mixpanel analytics tracking token for development work. Reach out to a lead to request a token.
+
+## MIXPANEL_PROD_TOKEN
+
+Mixpanel analytics tracking token for our live production site.
+
+## NEXT_PUBLIC_RSA_PRIVATE_KEY
+
+<!-- Introduced in #2096 but never documented. If you have information about this token, please contribute. Otherwise, this entry is flagged for removal, and the referrent code for refactoring. -->
+
+## NEXT_PUBLIC_RSA_PUBLIC_KEY
+
+<!-- Introduced in #2096 but never documented. If you have information about this token, please contribute. Otherwise, this entry is flagged for removal, and the referrent code for refactoring. -->
+
+## NO_CLIENT
+
+If `true`, disables the front-end build.
+
+## NO_GLOBAL_ACTIVITY_CACHE
+
+If `true`, disables the initialization of `globalActivityCache.ts` from server.
+
+## NO_PRERENDER
+
+In a production environment, prerender is only run from `commonwealth/server.ts` if this flag is false or blank.
+
+## NO_SSL
+
+Used and defined on-the-fly for the `start-external-webpack` package.json script.
+
+## NO_TOKEN_BALANCE_CACHE
+
+If `true`, the token balance cache is not initialized.
+
+## NODE_ENV
+
+The current environment; set to `production` or `development`. Upstream of how we handle several other env vars.
+
+## PGPASSWORD
+
+Postgres DB password (by default, `edgeware`). Bypasses usual password prompt for local DB commands.
+
+## PORT
+
+Localhost port location, default value `8080`.
+
+## PROCFILE
+
+Set in all of our Heroku apps that use a separate procfile. Specifies the path from the root of the repo to the Procfile to use for the deployed app. It is never needed locally but it must be set manually on Heroku and is required in production. Presently set to `packages/chain-events/Procfile`
+
+## RABBITMQ_API_URI
+
+<!-- In need of documentation; if you have information about this token, please contribute. -->
+
+## RABBITMQ_URI
+
+<!-- In need of documentation; if you have information about this token, please contribute. -->
+
+Should be set if running in the local environment.
+
+## REDIS_URL
+
+Location to host a local Redis instance. By default `redis://localhost:6379`.
+
+## REPEAT_TIME
+
+The number of minutes the ChainSubscriber should wait between requests to the CW app to retrieve the chains it should be listening to. It is currently set to `10`, meaning every 10 minutes the ChainSubscriber will query the CW App and retrieve all chains that have `has_chain_events_listener` set to True in the `Chains` model.
+
+## ROLLBAR_ENV
+
+A tag used to identify Rollbar reports by environment, such that we can easily filter on the Rollbar dashboard. In staging and demo apps, we set it to the name of the app. For local production, it should be set to some custom value such as your name in lowercase. It should NEVER be set to `production`.
```

Comment:
```suggestion
A tag used to identify Rollbar reports by environment, such that we can easily filter on the Rollbar dashboard. In staging and demo apps, we set it to the name of the app. Locally it should be set to some custom value such as your name in lowercase. It should NEVER be set to `production`.
```

Reply:
lmao, think I meant to type "local development," good catch